packages/netlify-cms-core/src/valueObjects/EditorComponent.js

```diff
@@ -28,8 +28,20 @@ export default function createEditorComponent(config) {
     type,
     icon,
     widget,
-    // enforce multiline flag, exclude others
-    pattern,
+    // We allow consumers to specify line anchors (^ and $) without a
+    // multiline flag, but we need to be able to match in a multi-line
+    // context. In order to do so, we replace instances of line anchors
+    // with lookarounds that include \n's, so that non-multiline expressions
+    // still work as expected.
+    //
+    // Note: This is mocked in packages/netlify-cms-widget-markdown/src/serializers/__tests__/remarkShortcodes.spec.js;
+    // if you change it here, change it there as well!
+    pattern: new RegExp(
```

Comment:
Would it make sense to import this function directly into the remarkShortcodes spec file to avoid allowing these overrides to get out of sync again? Is it safe to just add this is a dev dependency for netlify-cms-widget-markdown and import it?